kmcd-driver.md

```diff
@@ -20,25 +20,84 @@ module KMCD-DRIVER
           <msg-sender> 0:Address </msg-sender>
           <this> 0:Address </this>
           <currentTime> 0:Int </currentTime>
+          <callStack> .List </callStack>
+          <preState> .K </preState>
         </kmcd-driver>
 ```
 
 MCD Simulations
 ---------------
 
 ```k
-    syntax MCDStep
     syntax MCDSteps ::= ".MCDSteps" | MCDStep MCDSteps
  // --------------------------------------------------
     rule <k> .MCDSteps => . ... </k>
+    rule <k> MCD:MCDStep MCDS:MCDSteps => MCD ~> MCDS ... </k>
+
+    syntax MCDContract ::= contract(MCDStep) [function]
+    syntax Address ::= address(MCDContract) [function]
+ // ---------------------------------------------------
+```
+
+Function Calls
+--------------
+
+```k
+    syntax CallFrame ::= frame(prevSender: Address, continuation: K)
+
+    syntax AuthStep
+    syntax MCDStep ::= AuthStep
+    syntax MCDStep ::= "call" MCDStep
+ // ---------------------------------
+    rule <k> call AS:AuthStep ~> CONT => contract(AS) . auth ~> AS </k>
+         <msg-sender> MSGSENDER => THIS </msg-sender>
+         <this> THIS => address(contract(AS)) </this>
+         <callStack> .List => ListItem(frame(MSGSENDER, CONT)) ... </callStack>
+    rule <k> call MCD:MCDStep ~> CONT => MCD </k>
+         <msg-sender> MSGSENDER => THIS </msg-sender>
+         <this> THIS => address(contract(MCD)) </this>
+         <callStack> .List => ListItem(frame(MSGSENDER, CONT)) ... </callStack>
+      requires notBool isAuthStep(MCD)
+
+    syntax ReturnValue ::= Int | Rat
```

Comment:
Isn't `Int < Rat` anyway?

Reply:
It is but I think this is clearer conceptually.